crates/forge-runner/src/build_trace_data.rs

```diff
@@ -95,10 +99,14 @@ fn build_cairo_execution_info(
 fn get_source_sierra_path(
     contract_name: &str,
     contracts_data: &ContractsData,
-    test_artifacts_path: &Utf8Path,
+    maybe_test_program_path: &Option<TestProgramPath>,
 ) -> Option<Utf8PathBuf> {
     if contract_name == TEST_CODE_CONTRACT_NAME {
-        Some(Utf8PathBuf::from(test_artifacts_path))
+        Some(
+            maybe_test_program_path
+                .clone()
+                .map_or_else(Utf8PathBuf::new, Into::into),
```

Comment:
If the path is `None` then we won't be saving trace anyways, so it's meaningless

Reply:
So why is this function called then? This should be refactored

Reply:
If someone makes mistake in the code, you will potentially output an invalid trace (impossible to interpret)

Reply:
Yep, but this requires a lot more refactoring, creating an issue